common/server-datasource-ondexlocal/src/main/java/rres/knetminer/datasource/ondexlocal/OndexServiceProvider.java

```diff
@@ -676,28 +676,34 @@ private void mergeHits(HashMap<ONDEXConcept, Float> hit2score, ScoredHits<ONDEXC
          * "CHEMBLTARGET", "EC", "EMBL", "ENSEMBL", "GENB", "GENOSCOPE", "GO", "INTACT",
          * "IPRO", "KEGG", "MC", "NC_GE", "NC_NM", "NC_NP", "NLM", "OMIM", "PDB",
          * "PFAM", "PlnTFDB", "Poplar-JGI", "PoplarCyc", "PRINTS", "PRODOM", "PROSITE",
-         * "PUBCHEM", "PubMed", "REAC", "SCOP", "SOYCYC", "TAIR", "TX", "UNIPROTKB"};
+         * "PUBCHEM", "PubMed", "REAC", "SCOP", "SOYCYC", "TAIR", "TX", "UNIPROTKB", "UNIPROTKB-COV",
+         * "ENSEMBL-HUMAN"};
          */
         Set<String> dsAcc = new HashSet<>(Arrays.asList(datasources));
 
         HashMap<ONDEXConcept, Float> hit2score = new HashMap<>();
 
-        if ("".equals(keywords) || keywords == null) {
+	         if ("".equals(keywords) || keywords == null) {
             log.info("No keyword, skipping Lucene stage, using mapGene2Concept instead");
             if (geneList != null) {
                 for (ONDEXConcept gene : geneList) {
+                    log.info("gene is recorded as " + gene);
                     if (gene != null) {
-                        for (int conceptId : mapGene2Concepts.get(gene.getId())) {
-                            ONDEXConcept concept = graph.getConcept(conceptId);
-                            if (includePublications || !concept.getOfType().getId().equalsIgnoreCase("Publication")) {
-                                hit2score.put(concept, 1.0f);
+                        if (mapGene2Concepts.get(gene.getId()) != null) {
```

Comment:
Ususally, breaking/continuing when the current item doesn't neeed to be processed makes the code more compact and more readable, ie, 
```java
for (...)
{
  if ( !cond1 ) continue;
  if ( !cond2 ) continue;
  // else process it
}
```
As you can see, it avoids all the nesting.